python/jupytergis_lab/jupytergis_lab/notebook/gis_document.py

```diff
@@ -138,6 +143,56 @@ def _export_to_qgis(self, path: str | Path) -> bool:
 
         return export_project_to_qgis(path, virtual_file)
 
+    def add_layer(
```

Comment:
I don't know how feasible it will be in the long-run to have a function like this. My ideal is that the user shouldn't need to know what they have to explore it. This function has the potential to get pretty bloated as the number of supported layer types grows, but I think the cost is likely worth it. I don't think we need this function to support all of the layer types we support at this early stage. If users want it to support more things, they will ask for it! And maybe help add support too :)